azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java

```diff
@@ -81,34 +83,44 @@ private Location createLocation(final String input) {
       if (input == null) {
          return null;
       }
-      return new LocationBuilder().
-              id(input).
-              scope(LocationScope.REGION).
-              description(input).
-              parent(Iterables.getOnlyElement(provider.get())).
-              metadata(ImmutableMap.<String, Object>of("name", input)).
-              build();
+
+      final LocationBuilder builder = new LocationBuilder();
```

Comment:
Instead of manually creating a location here, you'd better inject the location supplier into this class and pick one of the existing locations.